src/components/base/BaseTag.vue

```diff
@@ -1,5 +1,22 @@
+<script setup lang="ts">
+import { defineProps } from 'vue';
+import { computed } from 'vue';
+
+const props = defineProps<{
+  theme?: 'gray' | 'red';
+}>();
```

Comment:
I've seen you handling the "default" value by falling back to red if the prop is undefined - why not define a default this way 😄 ?
```suggestion
const props = withDefaults(defineProps<{
  theme?: 'gray' | 'red';
}>(), {
  theme: 'red'
});
```

Reply:
Got it!

Also I've removed imports of `defineProps` since it's a macro available globally 😅